core/domain/topic_domain.py

```diff
@@ -155,16 +155,93 @@ def to_dict(self):
             'version': self.version
         }
 
+    @classmethod
+    def require_valid_topic_id(cls, topic_id):
+        """Checks whether the topic id is a valid one.
+
+        Args:
+            topic_id: str. The topic id to validate.
+        """
+        if not isinstance(topic_id, basestring):
+            raise utils.ValidationError(
+                'Topic id should be a string, received: %s' % topic_id)
+
+        if len(topic_id) != 12:
+            raise utils.ValidationError('Topic id %s is invalid' % topic_id)
+
+    @classmethod
+    def require_valid_name(cls, name):
+        """Checks whether the name of the topic is a valid one.
+
+        Args:
+            name: str. The name to validate.
+        """
+        if not isinstance(name, basestring):
+            raise utils.ValidationError('Name should be a string.')
+
+        if name == '':
+            raise utils.ValidationError('Name field should not be empty')
+
+    def delete_skill(self, skill_id):
+        """Removes a skill from the skill_ids list.
+
+        Args:
+            skill_id: str. The skill id to remove from the list.
+
+        Returns:
+            list. The updated list of skill ids.
+        """
+        skill_ids = self.skill_ids
```

Comment:
Not sure I understand why you need to return skill_ids. Why not just do `self.skill_ids.remove(skill_id)`?

Also might want to check if skill_id is there in the first place and raise exception if not.

Reply:
(same reason for all) I am using self.skill_ids for old value, do didn't want to change that.

Reply:
Ah, I don't think this will help. In Python lists are passed by reference. So you will be changing the old value anyway.

Instead, for old value, do a copy.deepcopy().

Reply:
Also in general if a dev needs to know that info to understand what's going on here, something is wrong -- the old_value stuff is in a distant part of the application. So in general something like this would explicitly merit a comment explaining the "why". But, better to do it in such a way that distant concerns are not tightly coupled in the first place.

Reply:
Done, added deepcopy in the service to calculate old_value before calling this function. Same for the 3 functions below.